qjs.c

```diff
@@ -164,6 +164,11 @@ static JSContext *JS_NewCustomContext(JSRuntime *rt)
     JSValue global = JS_GetGlobalObject(ctx);
     JS_SetPropertyFunctionList(ctx, global, global_obj, countof(global_obj));
     JS_SetPropertyFunctionList(ctx, global, &argv0, 1);
+    JSValue navigator = JS_NewObject(ctx);
+    char version[32];
+    snprintf(version, sizeof(version), "quickjs-ng/%s", JS_GetVersion());
+    JS_DefinePropertyValueStr(ctx, navigator, "userAgent", JS_NewString(ctx, version), JS_PROP_ENUMERABLE);
+    JS_DefinePropertyValueStr(ctx, global, "navigator", navigator, JS_PROP_C_W_E);
```

Comment:
For conformance reasons, these should be getters and `navigator` should be an instance of `Navigator`. I believe wpt tests for that and it's how browsers and node implement it (deno too, I believe; dunno 'bout bun)
```
> Object.getOwnPropertyDescriptor(globalThis, "navigator")
{
  get: [Function: get navigator],
  set: undefined,
  enumerable: true,
  configurable: true
}
> Object.getOwnPropertyDescriptor(navigator.__proto__, "userAgent")
{
  get: [Function: get userAgent],
  set: undefined,
  enumerable: true,
  configurable: true
}
```
Note the `__proto__` - `userAgent` is a prototype property, not an own property. `Navigator` can be either a getter or a value property on `globalThis`; node does the former, firefox the latter.

Reply:
I can tweak those! I wasn't shooting for full compliance here, but rather some alignment for user JS to be able to get more untime information in a somewhat standard way 😅